artemis/src/main/kotlin/ktx/artemis/entity.kt

```diff
@@ -0,0 +1,94 @@
+package ktx.artemis
+
+import com.artemis.Archetype
+import com.artemis.Component
+import com.artemis.EntityEdit
+import com.artemis.World
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.InvocationKind
+import kotlin.contracts.contract
+
+/**
+ * Creates and adds an entity to the [World].
+ *
+ * @receiver the [World] for creating the entity.
+ * @param entityEdit - the inlined function with [EntityEdit].
+ * @return the entity's id [Int].
+ */
+@OptIn(ExperimentalContracts::class)
+inline fun World.entity(entityEdit: EntityEdit.() -> Unit = {}): Int {
+  contract { callsInPlace(entityEdit, InvocationKind.EXACTLY_ONCE) }
+  val e = create()
```

Comment:
Thank you, and I fixed the variable names. I read the discussion about properties but I'm not sure if it's necessary. With artemis you can toggle an entity's component using a mapper e.g.:

```kotlin
visibilityMapper.set(entityId, true) // add (if missing)
visibilityMapper.set(entityId, false) // remove (if it exists)
```
There's also the [Fluid API](https://github.com/junkdog/artemis-odb/wiki/Fluid-Entities) which makes it quite easy to add and edit components. It does come with a performance cost though. But if you want the same features for consistency with `ktx-ashley` you can look into it.

Reply:
@deviodesign I think `entityId.visible = true` should be a little bit more natural in Kotlin than using mappers, but the fluid API does seem pretty nice. As I've said, I think this can be merged as is, and we can consider adding the extra properties later.